gtsfm/data_association/data_assoc.py

```diff
@@ -196,6 +204,68 @@ def run(
 
         return connected_data, data_assoc_metrics
 
+    def run_triangulation(
```

Comment:
What's the difference between `run_triangulation()` and `run_da()`? is one parallelized, and the other is not?

Technically, neither does any data association, since we broke DSF into another module, so can we use a different name than `run_da()`? (I know the module is mis-named currently)

Reply:
Technically it's still associating landmarks to keypoints. The class is also called DataAssociation, so we'd need to change the whole class name as well. Maybe we can change the naming conventions in another PR?

Reply:
I see, yeah I suppose it depends a bit on our definition of data association. Agreed it's orthogonal to this PR, but I still am a bit lost by the function naming, for the new function we're adding. Do we mean `run_da` vs. `run_da_batched`? or `run_triangulation` vs. `run_triangulation_batched`?

Reply:
Here's one definition of data association from Szeliski, that would suggest we rename this module:

In SFM, “data association” is the problem of determining correspondences, either between feature points or whole images. In the case of whole images, it can be seen as the validity of hypotheses that image pairs contain
sets of matching features corresponding to the same 3D points.

http://szeliski.org/papers/Roberts_DuplicateSfM_CVPR11.pdf

Reply:
(but of course module renaming can be for the future)

Reply:
Batching is only enforced for triangulation when the `MAX_DELAYED_TRIANGULATION_CALLS` is violated. I think a simple if-else statement is sufficient as opposed to a separate run function.

I think the function naming is straightforward. `run_triangulation` triangulates the 2D tracks. `run_da` performs the DA step (or at least what we were previously referring to as DA), i.e., validating the triangulated tracks and building the input to BA.

Yes, data association is the process of determining correspondences. This module still plays a role in determining correspondences, as we filter correspondences based on the triangulation geometry. 

You're more than welcome to rename this module to whatever you'd like in a separate PR :)